spring-integration-websocket/src/main/java/org/springframework/integration/websocket/inbound/WebSocketInboundChannelAdapter.java

```diff
@@ -324,6 +327,7 @@ else if (StompCommand.RECEIPT.equals(stompCommand)) {
 		}
 		else {
 			if (this.useBroker) {
+				Assert.notNull(this.brokerHandler, "brokerHandler is required");
```

Comment:
This is redundant.
We do have such a check in the `onInit()`.
This method has to be marked with a `@SuppressWarnings("NullAway") // Dataflow analysis limitation` instead.
See more info in docs: https://docs.spring.io/spring-framework/reference/7.0/core/null-safety.html#_warnings_suppression